components/channel_view/channel_view.jsx

```diff
@@ -48,13 +48,21 @@ export default class ChannelView extends React.PureComponent {
             updatedState = {...updatedState, deferredPostView: ChannelView.createDeferredPostView()};
```

Comment:
nit: why do we clone `updatedState` each time? It's already a new object, so should we just set the new properties as required?

Reply:
> However, I am concerned that we're just piling on tweaks to an already complex subsystem here. Would it make more sense to consider an architectural change to the `/unreads` endpoint that avoids it being coupled to the last unread timestamp in the first place?

I think this is how it should have been in a way. We should not be marking the channel as read without actually seeing the posts in the first place. Even here i am not making an exception incase the get posts fail(That i feel should be another ticket though as it kinda is a new behaviour).

Other thing is that the logic of marking previous channel as read makes this complex otherwise it should have been more obvious.
 
Let's see how this goes. If we are complicating this anymore we will definitely consider the suggestion of making a timestamp dependent API

Reply:
Good point it was already being cloned so i just copied it. I think we should change this here.

Reply:
I just changed this a bit. Though i removed clone at one condition i cannot remove on the other one as it can override the other one.